pkg/management/postgres/restore.go

```diff
@@ -398,6 +398,12 @@ func (info InitInfo) writeRestoreWalConfig(backup *apiv1.Backup, cluster *apiv1.
 			0o600)
 	}
 
+	if info.PgWal != "" {
+		if err := os.Symlink(info.PgWal, path.Join(info.PgData, "pg_wal")); err != nil {
```

Comment:
I think this will fail because Barman is going to create the `pg_wal` directory and `os.Symlink` is not going to overwrite it. Perhaps it's better to have a generic function that migrate the existing content of `pg_wal` inside  the volume, removes recursively the `pg_wal` directory, and create the symlink.

We could use this function when migrating an existing cluster which doesn't have a PVC for WALS to one having it.